src/main/java/frc/utils/calibration/sysid/SysIdCalibrator.java

```diff
@@ -1,108 +1,48 @@
 package frc.utils.calibration.sysid;
 
 import com.ctre.phoenix6.SignalLogger;
-import edu.wpi.first.units.Measure;
-import edu.wpi.first.units.Voltage;
+import edu.wpi.first.units.measure.Voltage;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
-import frc.utils.GBSubsystem;
+import frc.robot.subsystems.GBSubsystem;
 import frc.utils.joysticks.SmartJoystick;
-import org.littletonrobotics.junction.Logger;
 
 import java.util.function.Consumer;
 
-import static edu.wpi.first.units.Units.Seconds;
 import static edu.wpi.first.units.Units.Volts;
 
 public class SysIdCalibrator {
 
-	public static final double DEFAULT_VOLTAGE_STEP = 7;
-	public static final double DEFAULT_RAMP_RATE_VOLTS_PER_SECOND = 1;
-	public static final double DEFAULT_TIMEOUT_SECONDS = 10;
-
 	private final SysIdRoutine sysIdRoutine;
 	private final GBSubsystem usedSubsystem;
 	private final boolean isCTRE;
 
-	/**
-	 * IMPORTANT: You must do SignalLogger.stop() at the end of the calibration
-	 *
-	 * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
-	 */
-	public SysIdCalibrator(
-		boolean isCTRE,
-		GBSubsystem subsystem,
-		Consumer<Double> voltageSetControl,
-		double voltageStepVolts,
-		double rampRateVoltsPerSecond
-	) {
-		this(isCTRE, subsystem, voltageSetControl, voltageStepVolts, rampRateVoltsPerSecond, DEFAULT_TIMEOUT_SECONDS);
-	}
-
-	/**
-	 * IMPORTANT: You must do SignalLogger.stop() at the end of the calibration
-	 *
-	 * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
-	 */
-	public SysIdCalibrator(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl, double voltageStep) {
-		this(isCTRE, subsystem, voltageSetControl, voltageStep, DEFAULT_RAMP_RATE_VOLTS_PER_SECOND, DEFAULT_TIMEOUT_SECONDS);
-	}
-
-	/**
-	 * IMPORTANT: You must do SignalLogger.stop() at the end of the calibration
-	 *
-	 * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
-	 */
-	public SysIdCalibrator(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl) {
-		this(
-			isCTRE,
-			subsystem,
-			voltageSetControl,
-			DEFAULT_VOLTAGE_STEP,
-			DEFAULT_RAMP_RATE_VOLTS_PER_SECOND,
-			DEFAULT_TIMEOUT_SECONDS
-		);
-	}
+	public record SysIdConfigInfo(SysIdRoutine.Config config, boolean isCTRE) {}
```

Comment:
same about record inside another class

Reply:
Why not create another class file for it?

Reply:
i think its better here. the record meant for creation of this object and only this, like SysidRoutine.Config, SysidRoutine.Mechanism